### PR TITLE
fixes : #1177 support variable values on open libery alizer's port de…

### PR DESCRIPTION
**Pull request by Horiodino**
# Description of Changes
resolves the port value by checking if it is a variable and if it is, it returns the value of the variable

# Related Issue(s)
https://github.com/devfile/api/issues/1177

# Acceptance Criteria

_Testing and documentation do not need to be complete in order for this PR to be approved. However, tracking issues must be opened for missing testing/documentation._

- [ ] Unit/Functional tests

  

- [ ] Documentation

   

# Tests Performed
✔️️ Added suitable test case based on Issue.

# How To Test
_Instructions for the reviewer on how to test your changes._

# Notes To Reviewer
_Any notes you would like to include for the reviewer._

### CODE REVIEW

File: pkg/apis/enricher/framework/java/openliberty_detector_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		<httpEndpoint host="*" httpPort="${default.http.port}" httpsPort="${default.https.port}" id="defaultHttpEndpoint"/>
+	</server>
+	`
+	writeTestFile(t, tempDir, "variable_server.xml", variableXML)
+
+	detector := OpenLibertyDetector{}
+
+	ctx := context.TODO()
+
+	t.Run("Hardcoded Ports", func(t *testing.T) {
```
Comment (Jdubrick) [inline pkg/apis/enricher/framework/java/openliberty_detector_test.go:40]: Is it possible to add a test case for a mix of hardcoded and variable based ports?

Comment (Jdubrick) [reply pkg/apis/enricher/framework/java/openliberty_detector_test.go:40]: Possibly one for empty var as well?

**Review by Jdubrick — CHANGES_REQUESTED** [review]
Overall I like the changes :) just left 1 comment regarding adding a few more test cases to cover all scenarios

**Review by thepetk — APPROVED** [review]
/lgtm

File: go.mod [hunk 2]
```diff
@@ -17,6 +17,7 @@ require (
 	golang.org/x/mod v0.20.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
+	gotest.tools v1.4.0
```
Comment (Jdubrick) [inline go.mod:20]: Just need to run `go mod tidy` as @thepetk mentioned

**Review by Jdubrick — APPROVED** [review]
lgtm with the one comment about running `go mod tidy`

**Review by Jdubrick — APPROVED** [review]
lgtm

**Pull request merged**